server/common/config/dataset_config.py

```diff
@@ -0,0 +1,214 @@
+            self.app__authentication_enable = default_config["app"]["authentication_enable"]
+
+            self.presentation__max_categories = default_config["presentation"]["max_categories"]
+            self.presentation__custom_colors = default_config["presentation"]["custom_colors"]
+
+            self.user_annotations__enable = default_config["user_annotations"]["enable"]
+            self.user_annotations__type = default_config["user_annotations"]["type"]
+            self.user_annotations__local_file_csv__directory = default_config["user_annotations"]["local_file_csv"]["directory"]  # noqa E501
+            self.user_annotations__local_file_csv__file = default_config["user_annotations"]["local_file_csv"]["file"]
+            self.user_annotations__ontology__enable = default_config["user_annotations"]["ontology"]["enable"]
+            self.user_annotations__ontology__obo_location = default_config["user_annotations"]["ontology"]["obo_location"]  # noqa E501
+            self.user_annotations__hosted_tiledb_array__db_uri = default_config["user_annotations"]["hosted_tiledb_array"]["db_uri"]  # noqa E501
+            self.user_annotations__hosted_tiledb_array__hosted_file_directory = default_config["user_annotations"]["hosted_tiledb_array"]["hosted_file_directory"]  # noqa E501
+
+            self.embeddings__names = default_config["embeddings"]["names"]
+            self.embeddings__enable_reembedding = default_config["embeddings"]["enable_reembedding"]
+
+            self.diffexp__enable = default_config["diffexp"]["enable"]
+            self.diffexp__lfc_cutoff = default_config["diffexp"]["lfc_cutoff"]
+            self.diffexp__top_n = default_config["diffexp"]["top_n"]
+
+        except KeyError as e:
+            raise ConfigurationError(f"Unexpected config: {str(e)}")
+
+        # The annotation object is created during complete_config and stored here.
+        self.user_annotations = None
+
+    def complete_config(self, context):
+        self.handle_app()
+        self.handle_presentation()
+        self.handle_user_annotations(context)
+        self.handle_embeddings()
+        self.handle_diffexp(context)
+
+    def handle_app(self):
+        self.check_attr("app__scripts", list)
+        self.check_attr("app__inline_scripts", list)
+        self.check_attr("app__about_legal_tos", (type(None), str))
+        self.check_attr("app__about_legal_privacy", (type(None), str))
+        self.check_attr("app__authentication_enable", bool)
+
+        # scripts can be string (filename) or dict (attributes).   Convert string to dict.
+        scripts = []
+        for script in self.app__scripts:
+            try:
+                if isinstance(script, str):
+                    scripts.append({"src": script})
+                elif isinstance(script, dict) and isinstance(script["src"], str):
+                    scripts.append(script)
+                else:
+                    raise Exception
```

Comment:
Why throw the error later? Can you just raise the ConfigurationError here?

Reply:
Tried refactoring, but if script is a dict that doesnt contain `src` it will throw a key error which wont be caught